docs/oauth-setup.md

```diff
@@ -0,0 +1,143 @@
+# Using an external OAuth provider with S.H.I.E.L.D
+
+## Github
+
+### Open the Github settings page
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github1.png "github OAuth")
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github2.png "github OAuth")
+
+### Open "OAuth Apps" under the "Developer settings" field on the lower left menu
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github3.png "github OAuth")
+
+### Register a new Github OAuth client with the "Register a new application" button
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github4.png "github OAuth")
+
+- The only field that is SHIELD specific here is the "Authorization callback URL" field, which needs to be:
```

Comment:
This can be reworded to the shorter: "The `Authorization callback URL` field must be set to:"